### ISSUE TITLE
Training process with input init weight transforming

### ISSUE THREAD

**Issue by jinseok-karl** [zamponotiropita/Mask_RCNN#1]
Thanks to share this code!
I'd like to follow training process.
In my understanding, this code(paper) uses image-net pretrained weight which has 3 channel input structure.
So I planed my training process as follows, roughly pretrained weight manipulate -> scenenet train ->custom data finetune.
Specifically
1. Load mask-rcnn image net pretrained weight
2. use module instance_segmentation/sceneNet/train.py with init_with == "rgb_to_rgbd"
then it results 4-channel input weight.
3. Train with init_with == "sceneNet"
4. Train with custom data

So my issue is,  Is it right that init_with == "rgb_to_rgbd"  part is the weight structure transforming part?
And, nowadays, Imagenet weight which attained model.load_weights(model.get_imagenet_weights(), by_name=True) may not works.

**Comment by jinseok-karl** [comment]
Oh, It solved.
I don't know exactly where the code is, If you want to 4channel transfromed imagenet weight, you can load weight setting config as MODE = 'RGB' and IMAGE_SHAPE=[320 320  3].


    model.load_weights(model.get_imagenet_weights(), by_name=True)
    conv1 = model.keras_model.get_layer("conv1")
    kernel_rgb, bias = conv1.get_weights()

    # reload model in rgb-d mode
    config_rgbd = Config()
    config_rgbd.MODE = "RGBD"
    config_rgbd.MEAN_PIXEL = np.array([123.7, 116.8, 103.9, 255 / 2])
    config_rgbd.IMAGE_SHAPE = np.array([config.IMAGE_MAX_DIM, config.IMAGE_MAX_DIM, 4])
    config_rgbd.display()

    model_rgbd = modellib.MaskRCNN(mode="training", config=config_rgbd, model_dir=MODEL_DIR)
    model_rgbd.load_weights(MODEL_PATH, by_name=True, exclude=["conv1"])

    # set weights on first convolution from rgb model
    conv1 = model_rgbd.keras_model.get_layer("conv1")
    kernel_rgbd = np.concatenate((kernel_rgb, np.mean(kernel_rgb, keepdims=True, axis=2)), axis=2)
    conv1.set_weights([kernel_rgbd, bias])

    model = model_rgbd
    
   And save the weights.
Then is it okay to start with scene net?

**Comment by orestis-z** [comment]
Hi there

I load the weights from the first convolution in https://github.com/orestis-z/mask-rcnn-rgbd/blob/d590e0f5085f8cbe895a6698e284426fd0116aa4/instance_segmentation/sceneNet/train.py#L69 and then later on in https://github.com/orestis-z/mask-rcnn-rgbd/blob/d590e0f5085f8cbe895a6698e284426fd0116aa4/instance_segmentation/sceneNet/train.py#L84 I take the mean of the first 3 channels to initialize the weights for the depth channel.

Regarding which dataset to start with it highly depends on your training schedule. I did a pretraining on sceneNet and fine-tuning on NYU Depth V2

Let me know if you need anything else

**Comment by jinseok-karl** [comment]
Thanks to reply! To classify my problem step by step, I'll open another issue.
